roles/openshift_logging_elasticsearch/tasks/get_es_version.yml

```diff
@@ -1,23 +1,47 @@
 ---
-- command: >
-    {{ openshift_client_binary }} get pod -l component=es,provider=openshift -n {{ openshift_logging_elasticsearch_namespace }} -o jsonpath={.items[?(@.status.phase==\"Running\")].metadata.name}
-  register: _cluster_pods
+- oc_obj:
+    state: list
+    kind: pod
+    namespace: "{{ openshift_logging_elasticsearch_namespace }}"
+    selector: component=es
+  register: pod_list
+
+- set_fact:
+    available_pod: "{{ item.metadata.name }}"
+  with_items: "{{ pod_list.results.results[0]['items'] }}"
+  when:
+  - pod_list.results.results is defined
+  - item.status.phase == "Running"
+  - item.status.containerStatuses[0].ready == true
+  - item.status.containerStatuses[1].ready == true
 
 - name: "Getting ES version for logging-es cluster"
   command: >
-    {{ openshift_client_binary }} exec {{ _cluster_pods.stdout.split(' ')[0] }} -c elasticsearch -n {{ openshift_logging_elasticsearch_namespace }} -- {{ __es_local_curl }} -XGET 'https://localhost:9200/'
+    {{ openshift_client_binary }} exec {{ available_pod }} -c elasticsearch -n {{ openshift_logging_elasticsearch_namespace }} -- {{ __es_local_curl }} -XGET 'https://localhost:9200/'
   register: _curl_output
-  when: _cluster_pods.stdout_lines | count > 0
+  when: available_pod is defined
 
-- command: >
-    {{ openshift_client_binary }} get pod -l component=es-ops,provider=openshift -n {{ openshift_logging_elasticsearch_namespace }} -o jsonpath={.items[?(@.status.phase==\"Running\")].metadata.name}
-  register: _ops_cluster_pods
+- oc_obj:
+    state: list
+    kind: pod
+    namespace: "{{ openshift_logging_elasticsearch_namespace }}"
+    selector: component=es-ops
+  register: ops_pod_list
+
+- set_fact:
+    available_ops_pod: "{{ item.metadata.name }}"
+  with_items: "{{ ops_pod_list.results.results[0]['items'] }}"
+  when:
+  - ops_pod_list.results.results is defined
+  - item.status.phase == "Running"
+  - item.status.containerStatuses[0].ready == true
+  - item.status.containerStatuses[1].ready == true
 
 - name: "Getting ES version for logging-es-ops cluster"
   command: >
-    {{ openshift_client_binary }} exec {{ _ops_cluster_pods.stdout.split(' ')[0] }} -c elasticsearch -n {{ openshift_logging_elasticsearch_namespace }} -- {{ __es_local_curl }} -XGET 'https://localhost:9200/'
+    {{ openshift_client_binary }} exec {{ available_ops_pod }} -c elasticsearch -n {{ openshift_logging_elasticsearch_namespace }} -- {{ __es_local_curl }} -XGET 'https://localhost:9200/'
```

Comment:
If you don't mind relying on an image util
```
oc exec -c elasticsearch -n $NAMESPACE -- es_util --query=/
```
this give you the admin certs.  you can addtionally pass all curl args after 'query'

Reply:
lets take that up in another pr... i think there's more to clean up if we do that